packages/hooks/src/useTransferMessageBetweenWalletAndTradingApp.ts

```diff
@@ -0,0 +1,67 @@
+import useExchangeRate from './useExchangeRate';
+import useWalletTransfer from './useWalletTransfer';
+
+type TTransferMessageReturnType = {
+    code: 'WalletToTradingAppDailyLimit' | 'DemoWalletToTradingAppDailyLimit';
+    cta?: string;
+    is_first_transfer: boolean;
+    limit?: number;
+    currency?: string;
+    type: 'error' | 'info' | 'success';
+};
+
+const checkIsTradingAccount = (account_type = '') => {
+    if (account_type) {
+        const trading_account_types = ['ctrader', 'derivez', 'dxtrade', 'mt5', 'trading'];
+        return trading_account_types.includes(account_type);
+    }
+};
+
+const tradingAccountMapper = (account_type: string, currency_type: string): string => {
+    if (account_type === 'trading')
+        if (currency_type === 'demo') return 'virtual';
+        else return 'dtrade';
+    return account_type;
+};
```

Comment:
Need suggestions for caching checkIsTradingAccount and tradingAccountMapper the functions here.
cc: @heorhi-deriv @farzin-deriv 🙏
```suggestion
const checkIsTradingAccount = (account_type = '') => {
    if (account_type) {
        const trading_account_types = ['ctrader', 'derivez', 'dxtrade', 'mt5', 'trading'];
        return trading_account_types.includes(account_type);
    }
};

const tradingAccountMapper = (account_type: string, currency_type: string): string => {
    if (account_type === 'trading')
        if (currency_type === 'demo') return 'virtual';
        else return 'dtrade';
    return account_type;
};
```
```suggestion
const checkIsTradingAccount = (account_type = '') => {
    if (account_type) {
        const trading_account_types = ['ctrader', 'derivez', 'dxtrade', 'mt5', 'trading'];
        return trading_account_types.includes(account_type);
    }
};

const tradingAccountMapper = (account_type: string, currency_type: string): string => {
    if (account_type === 'trading')
        if (currency_type === 'demo') return 'virtual';
        else return 'dtrade';
    return account_type;
};
```

Reply:
@aum-deriv do we really need `checkIsTradingAccount` function?
instead of `if (from_account.account_type === 'wallet' && checkIsTradingAccount(to_account.account_type)) {`
can we do this:
if (from_account.account_type === 'wallet' && to_account.account_type !== 'wallet') {

Reply:
regarding caching:
we have useCallback and useMemo for caching
but here you already extract `tradingAccountMapper` to upper scope so the reference to the function will be the same every rerender.

or maybe i'm not fully understand your concern

Reply:
> regarding caching: we have useCallback and useMemo for caching but here you already extract `tradingAccountMapper` to upper scope so the reference to the function will be the same every rerender.
> 
> or maybe i'm not fully understand your concern

@heorhi-deriv, The thing is that this hook will be rerendered every time the `from_account` and `to_account` change as they are passed from its parent. So is there a need of caching them?

Reply:
This is resolved 👍